js/actions/index.ts

```diff
@@ -31,18 +32,39 @@ export const setCachingProgress = result => ({
     payload: result,
 });
 
+export const setFlowInformation = result => ({
+    type: 'FLOW_INFORMATION',
+    payload: result,
+});
+
 export const cachingProgress = (result) => {
     const progress = result.progress;
     const flowKey = result.flowKey;
+
     return (dispatch) => {
         if (progress === 100 && flowKey) {
-            manywho.model.addNotification(flowKey, {
-                message: 'Caching is complete. You are ready to go offline',
-                position: 'bottom',
-                type: 'success',
-                dismissible: true,
-            });
-            dispatch(setCachingProgress(0));
+            const errorPollingValues = 'An error caching data has occurred, your flow may not work as expected whilst offline';
+
+            pollForStateValues()
```

Comment:
I think that if we could make it so that we are not making the additional call to periodicallyPollForStateValues in Connection.ts, then we could simply just call periodicallyPollForStateValues here instead.

Reply:
Scenario 1 I think, yes, that is a separate topic. When an objectdata request fails, then   currently this is not handled (except for a crummy alert error). However, those objectdata requests that were successful do actually get their responses cached in indexdb. So really the UI should explain more accurately what has happened.

With scenario 2, what do think the implication of polling for state values only after any objectdata requests have completed would have here?

And scenario 3, when you change the network settings in devtools, the UI's offline state will eventually update on successful value poll, or if the flow makes a successful network request e.g invoke call etc.

I'm not to sure if my original review comment really impacts these scenarios that much. It is more just a case that an unnecessary call to the state/values endpoint is being made and that you might want to resolve that.

Reply:
the pollForStateValues also checks for the status of the network, it will switch to offline if the user is in offline mode, but also if the server can not be reached for any other reason, e.g. server is down, wrong network configuration, etc.

I think my question can be summarized in the follow questions:
do we need to check for network or values before the cache is fully processed (100%)?
do we need to check for network or values even if there isn't any data load?

Reply:
To answer Q1: I think no. The objectdata caching only happens when the flow is initializing/joining (which can only happen if there is an internet connection, this is one of the caveats of the offline solution, you have to make sure you initialize/join the flow when you have network connectivity.)

To answer Q2: Yes